src/auxiliary.jl

```diff
@@ -111,10 +114,17 @@ function get_coeff(a::TaylorN, v::Array{Int,1})
     get_coeff(a[order+1], v)
 end
 
-getindex(a::TaylorN, n::Int) = a.coeffs[n]
+function getindex(a::TaylorN, n::Int)
+    1 ≤ n ≤ length(a.coeffs) && return a.coeffs[n]
+    n ≤ get_order()+1 &&
+        return zero( HomogeneousPolynomial([zero(eltype(a))], n-1) )
```

Comment:
Maybe you are right. 

The subtlety is that for `a.order+1 < n <= get_order()` we can't rely on `a[n]` or `a.coeffs[n]`, since the length of `a.coeffs` is `a.order+1`.